### PR TITLE
Add emoji picker popup

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Also includes a couple improvements on chat in general:

No more focus outline.
Input box is now multiline and will grow.

TODO for a future ticket: the helper that pops up when you start typing `:she...`

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits f984aec29aef...3d4508a1b19c introduce visual changes on linux. <details><summary>:mag_right: 2 new, 5 changed</summary>

 * Added: **Markdown-emoji**  
   Markdown-emoji rendered
 * Added: **Chat_Input-Emoji_Open**  
   Chat_Input-Emoji_Open rendered
 * Changed: **unlock_folders-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479863155399-f984aec29aef-3d4508a1b19c-linux/unlock_folders-Normal-f984aec29aef.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479863155399-f984aec29aef-3d4508a1b19c-linux/unlock_folders-Normal-3d4508a1b19c.png)  
   unlock_folders-Normal comparison
 * Changed: **unlock_folders-No_paperkeys** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479863155399-f984aec29aef-3d4508a1b19c-linux/unlock_folders-No_paperkeys-f984aec29aef.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479863155399-f984aec29aef-3d4508a1b19c-linux/unlock_folders-No_paperkeys-3d4508a1b19c.png)  
   unlock_folders-No_paperkeys comparison
 * Changed: **Input-Long_Multiline_rows_Min_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479863155399-f984aec29aef-3d4508a1b19c-linux/Input-Long_Multiline_rows_Min_1-f984aec29aef.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479863155399-f984aec29aef-3d4508a1b19c-linux/Input-Long_Multiline_rows_Min_1-3d4508a1b19c.png)  
   Input-Long_Multiline_rows_Min_1 comparison
 * Changed: **Chat_Input-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479863155399-f984aec29aef-3d4508a1b19c-linux/Chat_Input-Normal-f984aec29aef.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479863155399-f984aec29aef-3d4508a1b19c-linux/Chat_Input-Normal-3d4508a1b19c.png)  
   Chat_Input-Normal comparison
 * Changed: **Chat_Input-Empty** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479863155399-f984aec29aef-3d4508a1b19c-linux/Chat_Input-Empty-f984aec29aef.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479863155399-f984aec29aef-3d4508a1b19c-linux/Chat_Input-Empty-3d4508a1b19c.png)  
   Chat_Input-Empty comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits f984aec29aef...3d4508a1b19c introduce visual changes on linux. <details><summary>:mag_right: 2 new, 3 changed</summary>

 * Added: **Markdown-emoji**  
   Markdown-emoji rendered
 * Added: **Chat_Input-Emoji_Open**  
   Chat_Input-Emoji_Open rendered
 * Changed: **Input-Long_Multiline_rows_Min_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479864446023-f984aec29aef-3d4508a1b19c-linux/Input-Long_Multiline_rows_Min_1-f984aec29aef.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479864446023-f984aec29aef-3d4508a1b19c-linux/Input-Long_Multiline_rows_Min_1-3d4508a1b19c.png)  
   Input-Long_Multiline_rows_Min_1 comparison
 * Changed: **Chat_Input-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479864446023-f984aec29aef-3d4508a1b19c-linux/Chat_Input-Normal-f984aec29aef.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479864446023-f984aec29aef-3d4508a1b19c-linux/Chat_Input-Normal-3d4508a1b19c.png)  
   Chat_Input-Normal comparison
 * Changed: **Chat_Input-Empty** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479864446023-f984aec29aef-3d4508a1b19c-linux/Chat_Input-Empty-f984aec29aef.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479864446023-f984aec29aef-3d4508a1b19c-linux/Chat_Input-Empty-3d4508a1b19c.png)  
   Chat_Input-Empty comparison
</summary>

File: desktop/renderer/renderer.html [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <meta charset="utf-8"/>
     <title>Keybase</title>
     <link rel="stylesheet" href="./style.css" />
+    <link rel="stylesheet" href="../node_modules/emoji-mart/css/emoji-mart.css" />
```
Comment (chrisnojima) [inline desktop/renderer/renderer.html]: i don't think this'll work w/ a release build?

Comment (chrisnojima) [inline desktop/renderer/renderer.html]: We could look into building the css w/ webpack...

File: shared/chat/conversation/input.desktop.js [hunk 2]
```diff
@@ -2,32 +2,70 @@
...
 import {globalColors, globalMargins, globalStyles} from '../../styles'
+import {Picker} from 'emoji-mart'
 
 import type {Props} from './'
 
-class Conversation extends Component<void, Props, void> {
-  _input: any
+type State = {
+  emojiPickerOpen: boolean,
+  inputText: ?string,
```
Comment (chrisnojima) [inline shared/chat/conversation/input.desktop.js]: why not just string and take out the || '' stuff below?

Comment (MarcoPolo) [inline shared/chat/conversation/input.desktop.js]: the '' is just for figuring out selection information, the `Input` tag does get a ?string

File: desktop/renderer/renderer.html [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <meta charset="utf-8"/>
+    <meta charset="utf-8" http-equiv="Content-Security-Policy" content="default-src filesystem: http://localhost:4000 https://keybase.io/; img-src https://keybase.io/ https://pbs.twimg.com/ https://avatars.githubusercontent.com/ https://images.coinbase.com/ https://s3.amazonaws.com/keybase_processed_uploads/ file:; style-src 'unsafe-inline'; script-src file: http://localhost:4000 'unsafe-eval' 'sha256-Zb8Tvqopq928KRdsoNdWXBvYYljHYxHPD6bx7PFlpEA='"/>
```
Comment (chromakode) [inline desktop/renderer/renderer.html]: What is `sha256-Zb8Tvqopq928KRdsoNdWXBvYYljHYxHPD6bx7PFlpEA=`?

**Pull request merged**